services/supervisor/src/lib.rs

```diff
@@ -97,23 +113,66 @@ fn update_configs(context: &mut CallContext<'_>, changes: Vec<ConfigChange>) {
                 );
             }
         }
+
+        ConfigChange::StartService(start_service) => {
+            log::trace!(
+                "Request add service with name {:?} from artifact {:?}",
+                start_service.name,
+                start_service.artifact
+            );
+            if let Err(e) = context.start_adding_service(
+                start_service.artifact,
+                start_service.name,
+                start_service.config,
+            ) {
+                // Panic will cause changes to be rolled back.
+                panic!(
```

Comment:
Maybe it makes sense to change signature of the `before_commit`, so services will be able to return `Err(..)` instead of panicking.